python/ray/experimental/dag/actor_node.py

```diff
@@ -0,0 +1,117 @@
+import ray
+from ray.experimental.dag.dag_node import DAGNode
+
+from typing import Any, Dict, List, Optional, Tuple
+
+
+class ActorNode(DAGNode):
+    """Represents an actor creation in a Ray task DAG."""
+
+    def __init__(self, actor_cls: type, cls_args, cls_kwargs, cls_options):
+        self._actor_cls = actor_cls
+        self._last_call: Optional["ActorMethodNode"] = None
+        DAGNode.__init__(self, cls_args, cls_kwargs, cls_options)
+
+    def _copy_impl(
+        self,
+        new_args: List[Any],
+        new_kwargs: Dict[str, Any],
+        new_options: Dict[str, Any],
+    ):
+        return ActorNode(self._actor_cls, new_args, new_kwargs, new_options)
+
+    def _execute_impl(self):
+        if self._bound_options:
+            return (
+                ray.remote(self._actor_cls)
+                .options(**self._bound_options)
+                .remote(*self._bound_args, **self._bound_kwargs)
+            )
+        else:
+            return ray.remote(self._actor_cls).remote(
+                *self._bound_args, **self._bound_kwargs
+            )
+
+    def __getattr__(self, method_name: str):
+        # Raise an error if the method is invalid.
+        getattr(self._actor_cls, method_name)
+        call_node = _UnboundActorMethodNode(self, method_name)
```

Comment:
Is there a reason we do this lazily? Why not just raise an error right here?

Reply:
The error is raised immediately on line 37, it's not lazy.